src/pages/settings/PreferencesPage.js

```diff
@@ -38,42 +48,74 @@ const priorityModes = {
     },
 };
 
-const PreferencesPage = ({priorityMode}) => (
-    <>
-        <HeaderGap />
-        <HeaderWithCloseButton
-            title="Preferences"
-            shouldShowBackButton
-            onBackButtonPress={() => redirect(ROUTES.SETTINGS)}
-            onCloseButtonPress={() => redirect(ROUTES.HOME)}
-        />
-        <View style={styles.pageWrapper}>
-            <View style={[styles.settingsPageBody, styles.mb6]}>
-                <Text style={[styles.formLabel]} numberOfLines={1}>
-                    Priority Mode
-                </Text>
-                <View style={[styles.mb2]}>
-                    {/* empty object in placeholder below to prevent default */}
-                    {/* placeholder from appearing as a selection option. */}
-                    <RNPickerSelect
-                        onValueChange={
-                            mode => NameValuePair.set(CONST.NVP.PRIORITY_MODE, mode, ONYXKEYS.PRIORITY_MODE)
-                        }
-                        items={Object.values(priorityModes)}
-                        style={styles.picker}
-                        useNativeAndroidPickerStyle={false}
-                        placeholder={{}}
-                        value={priorityMode}
-                        Icon={() => <Icon src={DownArrow} />}
-                    />
+class PreferencesPage extends Component {
```

Comment:
Hey, so this was the direction I went down to follow the "offline first" approach. Basically if it was going purely based off the `user.expensifyNewsStatus` then it would seem unresponsive/slow for users with no or slow connections. I guess a nice follow up to handle the errors would be to show a toast & reset back to the initial onyx value but I believe that's out of scope.

I can also revert it back to a func component and go solely off the prop if needed.

Reply:
Hmm I see that does make sense now but that makes me question whether the `User.setExpensifyNewsStatus` should first be setting the data in onyx and then calling the API.

cc @tgolen what are your thoughts on this?

Reply:
I like the idea of optimistically setting the value in Onyx before calling the API, yeah. Then if the API fails, the Onyx data can be reset.

Reply:
Hmm, I'm not sure I agree with this, as I think we're going to end up with a lot of duplicated code with having to handle the case where we don't get a response at all and when we get a failed response (see [here](https://github.com/Expensify/Expensify.cash/pull/1716/files#diff-93c1bb3a403f942b2809d43de01f06366a9fed3060e3894959c268b53eb97be8R70-R78)). I get that we want to keep that offline first functionality though, so I'm not really sure which is the best route to go. 

@marcaaron this didn't get brought up when I first added `User.setExpensifyNewsStatus` so I'm curious to see if you have opinions on this.

Reply:
I like the optimistic approach here as well. Only thought is that we should consider updating the value of this data with the data returned from the _response_ instead of relying purely on response codes - but it's probably OK with this particular setting being a boolean and all.

I'm also inclined to wait and see how many examples like this we end up with (though I agree it will be a lot eventually). Would love to see a more elegant or baked in approach to this optimistic response pattern soon!